firebase-firestore/src/main/java/com/google/firebase/firestore/core/FieldFilter.java

```diff
@@ -64,20 +64,24 @@ public static FieldFilter create(FieldPath path, Operator operator, Value value)
     if (path.isKeyField()) {
       if (operator == Operator.IN) {
         return new KeyFieldInFilter(path, value);
+      } else if (operator == Operator.NOT_IN) {
+        return new KeyFieldNotInFilter(path, value);
       } else {
         hardAssert(
             operator != Operator.ARRAY_CONTAINS && operator != Operator.ARRAY_CONTAINS_ANY,
             operator.toString() + "queries don't make sense on document keys");
         return new KeyFieldFilter(path, operator, value);
       }
     } else if (Values.isNullValue(value)) {
-      if (operator != Filter.Operator.EQUAL) {
+      // TODO(ne-queries): Update error message to include != operator.
+      if (operator != Filter.Operator.EQUAL && operator != Filter.Operator.NOT_EQUAL) {
```

Comment:
`Operator` must already be imported here (see references to `Operator.ARRAY_CONTAINS`) so you can drop the `Filter` in `Filter.Operator`.